xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecFFmpeg.cpp

```diff
@@ -1188,11 +1188,15 @@ IHardwareDecoder* CDVDVideoCodecFFmpeg::CreateVideoDecoderHW(AVPixelFormat pixfm
 #define VP_VIDEOCODEC_HW
 IHardwareDecoder* CDVDVideoCodecFFmpeg::CreateVideoDecoderHW(AVPixelFormat pixfmt, CProcessInfo &processInfo)
 {
+#if defined(HAVE_LIBVA)
```

Comment:
see here: https://github.com/xbmc/xbmc/pull/10918
cmake looks for platform specific headers. this header file defines VP_VIDEOCODEC_HW (see some lines above) and bings new decoders to the factory.
my vision is that platform dependent code can be maintained outside of mainline and new ports can be implemented by adding files to a build. then there won't be any need for having ports like IMX in kodi's repository.